tests/framework/filters/AccessRuleTest.php

```diff
@@ -88,7 +129,55 @@ public function testMatchAction()
 
     // TODO test match controller
 
-    // TODO test match roles
+    public function testMatchRole()
+    {
```

Comment:
Current unit test for the ACF are built up like this. I doubt I'll have the time now to rewrite the whole thing now. Please take a look at the complete test:
https://github.com/yiisoft/yii2/blob/master/tests/framework/filters/AccessRuleTest.php